Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.ps1

```diff
@@ -26,36 +26,26 @@ function UpdateIntegrationContext([OAuth2DeviceCodeClient]$client){
 }
 
 function CreateNewSession {
-    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingConvertToSecureStringWithPlainText', '', Scope='Function')]
-    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingPlainTextForPassword', '', Scope='Function')]
-    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '', Scope='Function')]
-    param([string]$url, [string]$upn, [string]$password, [string]$bearer_token, [bool]$insecure, [bool]$proxy)
-    if ($password){
-        $url = "$url/powershell-liveid"
-    }
-    else
-    {
-        $url = "$url/powershell-liveid?BasicAuthToOAuthConversion=true"
-    }
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingConvertToSecureStringWithPlainText', '', Scope = 'Function')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingPlainTextForPassword', '', Scope = 'Function')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '', Scope = 'Function')]
+    param([string]$url, [string]$upn, [string]$bearer_token, [bool]$insecure, [bool]$proxy)
+    $url = "$url/powershell-liveid?BasicAuthToOAuthConversion=true"
+    $credential = ConvertTo-SecureString "Bearer $bearer_token" -AsPlainText -Force
 
-    if ($password){
-        $credential = ConvertTo-SecureString "$password" -AsPlainText -Force
-    } else {
-        $credential = ConvertTo-SecureString "Bearer $bearer_token" -AsPlainText -Force
-    }
     $credential = New-Object System.Management.Automation.PSCredential($upn, $credential)
     $session_option_params = @{
         "SkipCACheck" = $insecure
         "SkipCNCheck" = $insecure
     }
-    $session_options =  New-PSSessionOption @session_option_params
+    $session_options = New-PSSessionOption @session_option_params
     $sessions_params = @{
         "ConfigurationName" = "Microsoft.Exchange"
-        "ConnectionUri" = $url
-        "Credential" = $credential
-        "Authentication" = "Basic"
-        "AllowRedirection" = $true
-        "SessionOption" = $session_options
+        "ConnectionUri"     = $url
+        "Credential"        = $credential
+        "Authentication"    = "Basic"
+        "AllowRedirection"  = $true
+        "SessionOption"     = $session_options
     }
     $session = New-PSSession @sessions_params -WarningAction:SilentlyContinue
```

Comment:
It was decided that this is out-of-scope for the initial bug, and it will be assigned to a separate enhancement task.
There's a separate ticket for that on [CIAC-6282](https://jira-hq.paloaltonetworks.local/browse/CIAC-6282).